### PR TITLE
improve input.md

### PR DESCRIPTION
**Pull request by ggabernet**
Including docs on how to restart the pipeline from later steps in `input.md`. Improving a bit the docs.

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/sarek branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/sarek)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [x] Documentation in `docs` is updated
- [x] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by ggabernet** [comment]
@mGauder do you wanna check if it is a bit clearer now?

File: docs/input.md [hunk 1]
```diff
@@ -6,30 +6,30 @@ Input files for Sarek can be specified using a TSV file given to the `--input` c
 The TSV file is a Tab Separated Value file with columns:
 
 - `subject sex status sample lane fastq1 fastq2` for step `mapping` with paired-end FASTQs
-- `subject sex status sample lane bam` for step `mapping` with unmapped BAMs
-- `subject sex status sample bam bai recaltable` for step `recalibrate` with BAMs
+- `subject sex status sample lane bam` for step `mapping` with unmapped BAMs (uBAMs)
+- `subject sex status sample bam bai recaltable` for step `recalibrate` with mapped BAMs
```
Comment (MaxUlysse) [inline docs/input.md:10]:
```suggestion
- `subject sex status sample bam bai recaltable` for step `recalibrate` with mapped BAMs and corresponding recalibration table
```

File: docs/input.md [hunk 2]
```diff
@@ -6,30 +6,30 @@ Input files for Sarek can be specified using a TSV file given to the `--input` c
...
 
-- `subject` designate the subject, it should be the ID of the Patient, and it must design only one patient
+- `subject` designates the subject, it should be the ID of the Patient, and it must be unique for each patient
 - `sex` are the sex chromosomes of the Patient, (XX or XY)
-- `status` is the status of the Patient, (0 for Normal or 1 for Tumor)
-- `sample` designate the Sample, it should be the ID of the sample (it is possible to have more than one tumor sample for each patient, i.e. a tumor and a relapse), it must design only one sample
-- `lane` is used when the sample is multiplexed on several lanes, it must be unique for each lane in the same sample
+- `status` is the status of the measured sample, (0 for Normal or 1 for Tumor)
+- `sample` designates the Sample, it should be the ID of the sample (it is possible to have more than one tumor sample for each patient, i.e. a tumor and a relapse), it must be unique for each sample
+- `lane` is used when the sample is multiplexed on several lanes, it must be unique for each lane in the same sample (but does not need to be the original lane name), and must contain at least one letter
```
Comment (MaxUlysse) [inline docs/input.md:19]:
```suggestion
- `lane` is used when the sample is multiplexed on several lanes, it must be unique for each lane in the same sample (but does not need to be the original lane name), and must contain at least one character
```

File: docs/input.md [hunk 3]
```diff
@@ -117,6 +117,8 @@ G15511    XX    0    C09DFN    pathToFiles/G15511.C09DFN.md.bam    pathToFiles/G
 G15511    XX    1    D0ENMT    pathToFiles/G15511.D0ENMT.md.bam    pathToFiles/G15511.D0ENMT.md.bai pathToFiles/G15511.D0ENMT.md.recal.table
 ```
 
+When starting Sarek from the mapping step, a TSV file is generated automatically after the MarkDuplicates process. This TSV file is stored under `results/Preprocessing/TSV/duplicateMarked.tsv` and can be used to restart Sarek from the non-recalibrated BAM files, giving it as `--input` and setting the step `--step recalibrate`.
```
Comment (MaxUlysse) [inline docs/input.md:120]:
```suggestion
When starting Sarek from the mapping step, a TSV file is generated automatically after the `MarkDuplicates` process. This TSV file is stored under `results/Preprocessing/TSV/duplicateMarked.tsv` and can be used to restart Sarek from the non-recalibrated BAM files, giving it as `--input` and setting the step `--step recalibrate`.
```

Comment (MaxUlysse) [reply docs/input.md:120]: Actually, if you do just `--step recalibrate` sarek should fetch the correct input file

File: docs/input.md [hunk 4]
```diff
@@ -126,6 +128,8 @@ G15511    XX    0    C09DFN    pathToFiles/G15511.C09DFN.md.recal.bam    pathToF
 G15511    XX    1    D0ENMT    pathToFiles/G15511.D0ENMT.md.recal.bam    pathToFiles/G15511.D0ENMT.md.recal.bai
 ```
 
+When starting Sarek from the mapping or recalibrate steps, a TSV file is generated automatically after the recalibration processes. This TSV file is stored under `results/Preprocessing/TSV/recalibrated.tsv` and can be used to restart Sarek from the recalibrated BAM files, giving it as `--input` and setting the step `--step variantcalling`.
```
Comment (MaxUlysse) [inline docs/input.md:131]: Same here with `--step variantcalling`

File: docs/input.md (hunk 3 above)
Comment (MaxUlysse) [reply docs/input.md:120]: We could add that a TSV file is generated for each sample as well

File: docs/input.md (hunk 4 above)
Comment (MaxUlysse) [reply docs/input.md:131]: Same here: we could add that a TSV file is generated for each sample as well

Comment (ggabernet) [reply docs/input.md:131]: ok, great, I'll add this as well

File: docs/input.md [hunk 5]
```diff
@@ -1,41 +1,47 @@
...
-The content of these columns is quite straight-forward:
-
-- `subject` designate the subject, it should be the ID of the Patient, and it must design only one patient
+- `subject` designates the subject, it should be the ID of the Patient, and it must be unique for each patient
 - `sex` are the sex chromosomes of the Patient, (XX or XY)
-- `status` is the status of the Patient, (0 for Normal or 1 for Tumor)
-- `sample` designate the Sample, it should be the ID of the sample (it is possible to have more than one tumor sample for each patient, i.e. a tumor and a relapse), it must design only one sample
-- `lane` is used when the sample is multiplexed on several lanes, it must be unique for each lane in the same sample
+- `status` is the status of the measured sample, (0 for Normal or 1 for Tumor)
+- `sample` designates the Sample, it should be the ID of the sample (it is possible to have more than one tumor sample for each patient, i.e. a tumor and a relapse), it must be unique for each sample
```
Comment (MaxUlysse) [inline docs/input.md:12]: Actually if the sample is run on several lanes, then they do need to have the same sample ID to be merged.

Comment (MaxUlysse) [inline docs/input.md:12]:
```suggestion
- `sample` designates the sample, it should be the ID of the sample (it is possible to have more than one tumor sample for each patient, i.e. a tumor and a relapse), it must be unique for each sample
```

File: docs/input.md [hunk 6]
```diff
@@ -1,41 +1,47 @@
...
+- `lane` is used when the sample is multiplexed on several lanes, it must be unique for each lane in the same sample (but does not need to be the original lane name), and must contain at least one character
 - `fastq1` is the path to the first pair of the fastq file
 - `fastq2` is the path to the second pair of the fastq file
 - `bam` is the bam file
```
Comment (MaxUlysse) [inline docs/input.md:16]:
```suggestion
- `bam` is the path to the bam file
```

File: docs/input.md [hunk 7]
```diff
@@ -1,41 +1,47 @@
...
 - `bai` is the bam index file
```
Comment (MaxUlysse) [inline docs/input.md:17]:
```suggestion
- `bai` is the path to the bam index file
```

File: docs/input.md [hunk 8]
```diff
@@ -1,41 +1,47 @@
...
 - `recaltable` is the recalibration table
```
Comment (MaxUlysse) [inline docs/input.md:18]:
```suggestion
- `recaltable` is the path to the recalibration table
```

File: docs/input.md (hunk 5 above)
Comment (ggabernet) [reply docs/input.md]: right, but still the sample ID should be unique for each sample. I will add this info in the `lane` description though.

Comment (ggabernet) [reply docs/input.md]: I don't know what to change there actually, it looks pretty clear to me already, but of course open to suggestions :)

File: docs/input.md [hunk 9]
```diff
@@ -1,41 +1,47 @@
...
-The content of these columns is quite straight-forward:
-
-- `subject` designate the subject, it should be the ID of the Patient, and it must design only one patient
+- `subject` designates the subject, it should be the ID of the Patient, and it must be unique for each patient
 - `sex` are the sex chromosomes of the Patient, (XX or XY)
-- `status` is the status of the Patient, (0 for Normal or 1 for Tumor)
-- `sample` designate the Sample, it should be the ID of the sample (it is possible to have more than one tumor sample for each patient, i.e. a tumor and a relapse), it must design only one sample
-- `lane` is used when the sample is multiplexed on several lanes, it must be unique for each lane in the same sample
+- `status` is the status of the measured sample, (0 for Normal or 1 for Tumor)
+- `sample` designates the sample, it should be the ID of the sample (it is possible to have more than one tumor sample for each patient, i.e. a tumor and a relapse), it must be unique for each sample
```
Comment (MaxUlysse) [inline docs/input.md:12]:
```suggestion
- `sample` designates the sample, it should be the ID of the sample (it is possible to have more than one tumor sample for each patient, i.e. a tumor and a relapse), it must be unique, but samples can have multiple lanes (which will later be merged)
```

Comment (MaxUlysse) [reply docs/input.md:12]: What about that?

File: docs/input.md [hunk 10]
```diff
@@ -1,41 +1,47 @@
...
 
-- `subject sex status sample lane fastq1 fastq2` for step `mapping` with paired-end FASTQs
-- `subject sex status sample lane bam` for step `mapping` with unmapped BAMs
-- `subject sex status sample bam bai recaltable` for step `recalibrate` with BAMs
-- `subject sex status sample bam bai` for step `variantcalling` with BAMs
-
-The content of these columns is quite straight-forward:
-
-- `subject` designate the subject, it should be the ID of the Patient, and it must design only one patient
+- `subject` designates the subject, it should be the ID of the Patient, and it must be unique for each patient
```
Comment (ggabernet) [inline docs/input.md:9]:
```suggestion
- `subject` designates the subject, it should be the ID of the patient, and it must be unique for each patient, but one patient can have multiple samples (e.g. normal and tumor)
```

**Comment by ggabernet** [comment]
awesome, I think everything is covered now 😄

**Comment by ggabernet** [comment]
I would wait for @mGauder feedback, she's back tomorrow, to have another pair of eyes.

**Comment by MaxUlysse** [comment]
OK, I'll wait her approval to merge then ;-)

**Pull request merged**